scripts/choco-install.ps1

```diff
@@ -0,0 +1,38 @@
+$nodejs = "nodejs.install"
+$python = "python3"
+
+# Check for existence of required environment variables
+if ( $env:ChocolateyInstall -eq $null ) {
+  [Console]::Error.WriteLine('Missing $env:ChocolateyInstall environment variable')
+  exit 1
+}
+
+# Add the cached packages with source priority 1 (Chocolatey community is 0)
+New-Item -Path "${PSScriptRoot}\..\tmp\chocolatey" -ItemType "directory" -ErrorAction:SilentlyContinue
+choco source add --name="cache" --source="${PSScriptRoot}\..\tmp\chocolatey" --priority=1
+
+# Install nodejs v16.14.2 (will use cache if exists)
+choco install "$nodejs" --version="16.14.2" --require-checksums -y
+# Internalise nodejs to cache if doesn't exist
+if ( -not (Test-Path -Path "${PSScriptRoot}\..\tmp\chocolatey\$nodejs\$nodejs.16.14.2.nupkg" -PathType Leaf) ) {
+  Rename-Item -Path "$env:ChocolateyInstall\lib\$nodejs\$nodejs.nupkg" -NewName "$nodejs.nupkg.zip" -ErrorAction:SilentlyContinue
+  Expand-Archive -LiteralPath "$env:ChocolateyInstall\lib\$nodejs\$nodejs.nupkg.zip" -DestinationPath "$env:ChocolateyInstall\lib\$nodejs" -Force
+  Remove-Item "$env:ChocolateyInstall\lib\$nodejs\_rels" -Recurse
+  Remove-Item "$env:ChocolateyInstall\lib\$nodejs\package" -Recurse
+  Remove-Item "$env:ChocolateyInstall\lib\$nodejs\[Content_Types].xml"
+  New-Item -Path "${PSScriptRoot}\..\tmp\chocolatey\$nodejs" -ItemType "directory" -ErrorAction:SilentlyContinue
+  choco pack "$env:ChocolateyInstall\lib\$nodejs\$nodejs.nuspec" --outdir "${PSScriptRoot}\..\tmp\chocolatey\$nodejs"
+}
```

Comment:
This could be a function that is shared between the 2 invocations. The only difference is the package name.

Reply:
This logic is now separated out into its own function `Save-ChocoPackage` (that specific name was chosen to align with PowerShell function naming conventions https://docs.microsoft.com/en-us/powershell/scripting/learn/ps101/09-functions?view=powershell-7.2)